src/server/pfs/drive/driver.go

```diff
@@ -488,6 +488,9 @@ func (d *driver) MakeDirectory(file *pfs.File, shard uint64) (retErr error) {
 		}
 		diffInfo.Appends[path.Clean(file.Path)] = _append
 	}
+	// The fact that this is a directory is signified by setting Children
+	// to non-nil
+	_append.Children = make(map[string]bool)
```

Comment:
Basically, imagine that in a commit, you do `rm /pfs/out/dir && mkdir /pfs/out/dir`.  Without this change, we end up with an append with `Delete` set to true and `Children` being nil.  But looking at the append, we wouldn't be able to tell whether there is a directory there, or if the file has simply been deleted.  Having `Children` being an empty map (as opposed to nil) tells us that a directory has been created here.

So yes, the changes require this, because the changes now allow us to delete a file and create the file (or directory) in the same commit.